### PR TITLE
Branding custom texts :  Add support to preview multiple variations of the same screen

### PR DESCRIPTION
**Pull request by RushanNanayakkara**
### Purpose
The branding feature currently supports text customisation for selected screens. However, when the screen is dynamic and has multiple variations, it's not possible to preview these multiple variations of the screen. 
`SMS OTP for password recovery` feature requires the multi screen support for branding. 
This PR adds the functionality to preview variations for the selected screen through the preview panel. 

### Related Issues
- https://github.com/wso2/product-is/issues/20243

### Related PRs
- 

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [x] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### Screen recording 

https://github.com/wso2/identity-apps/assets/42939752/9dc1374c-6b2c-42e1-8471-47a5f4136444

### CODE REVIEW

File: features/admin.branding.v1/components/screen-variation-dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
```
Comment (RushanNanayakkara) [inline features/admin.branding.v1/components/screen-variation-dropdown.tsx:5]: Hide the drop down if there are no variations specified.

Comment (RushanNanayakkara) [reply features/admin.branding.v1/components/screen-variation-dropdown.tsx:5]: or if the variation has only one option.

File: features/admin.branding.v1/components/screen-variation-dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import React, { FunctionComponent, ReactElement, useEffect, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { DropdownProps, Form, Select } from "semantic-ui-react";
+import useBrandingPreference from "../hooks/use-branding-preference";
+import { PreviewScreenType, PreviewScreenVariationType } from "../models/branding-preferences";
+import { BASE_DISPLAY_VARIATION } from "../models/custom-text-preference";
+
```
Comment (RushanNanayakkara) [inline features/admin.branding.v1/components/screen-variation-dropdown.tsx:26]: remove extra empty line. keep one

File: features/admin.branding.v1/components/screen-variation-dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    const { getScreenVariations } = useBrandingPreference();
+
+    const { t } = useTranslation();
+
+    const [ screenVariations, setScreenVariations ] = useState<PreviewScreenVariationType[]>();
+
+    const [ selectedScreenVariation, setSelectedScreenVariation ]
+        = useState<PreviewScreenVariationType>();
+
+    useEffect(() => {
```
Comment (RushanNanayakkara) [inline features/admin.branding.v1/components/screen-variation-dropdown.tsx:70]: Add doc comments for the funcitons and use effects

File: features/admin.branding.v1/context/branding-preference-context.tsx [hunk 4]
```diff
@@ -39,6 +40,15 @@ export interface BrandingPreferenceContextProps {
      * @param screen - Selected screen.
      */
     onSelectedPreviewScreenChange: (screen: PreviewScreenType) => void;
+    /**
+     * Sets the variation of the current screen, requested for preview.
+     * @param variation - Selected screen variation.
+     */
+    onSelectedPreviewScreenVariationChange: (variation: PreviewScreenVariationType) => void;
+    /**
+     * Resets the variations back to default
```
Comment (RushanNanayakkara) [inline features/admin.branding.v1/context/branding-preference-context.tsx:49]: full stop at the end

File: .gitignore [hunk 5]
```diff
@@ -134,3 +134,5 @@ tests/dist/
 .repository/
 apps/console/src/extensions/i18n/tmp/**
 .nx
+apps/myaccount/v8-compile-cache-0/**
```
Comment (RushanNanayakkara) [inline .gitignore:137]: remove this and not add the files to the repo as well. these are temp files

File: features/admin.branding.v1/components/screen-variation-dropdown.tsx (hunk 1 above)
Comment (RushanNanayakkara) [reply features/admin.branding.v1/components/screen-variation-dropdown.tsx:5]: addressed with https://github.com/wso2/identity-apps/pull/6027/commits/f4caedf5091ff4248069acafa3599ddc0f886d53

File: features/admin.branding.v1/components/screen-variation-dropdown.tsx (hunk 2 above)
Comment (RushanNanayakkara) [reply features/admin.branding.v1/components/screen-variation-dropdown.tsx:26]: addressed with https://github.com/wso2/identity-apps/pull/6027/commits/6f514ee32c24be59d56b0f82cd4c13abfa87c27b

File: features/admin.branding.v1/context/branding-preference-context.tsx (hunk 4 above)
Comment (RushanNanayakkara) [reply features/admin.branding.v1/context/branding-preference-context.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6027/commits/6f514ee32c24be59d56b0f82cd4c13abfa87c27b

File: features/admin.branding.v1/components/screen-variation-dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+ * @returns Screen dropdown component.
+ */
+const ScreenVariationDropdown: FunctionComponent<ScreenVariationDropdownPropsInterface> = (
+    props: ScreenVariationDropdownPropsInterface
+): ReactElement => {
+    const {
+        onChange,
+        selectedScreen,
+        required,
+        ["data-componentid"]: componentId } = props;
```
[the remark this answers is not in the document]
Comment (RushanNanayakkara) [reply features/admin.branding.v1/components/screen-variation-dropdown.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6027/commits/6f514ee32c24be59d56b0f82cd4c13abfa87c27b

File: features/admin.branding.v1/components/screen-variation-dropdown.tsx (hunk 3 above)
Comment (RushanNanayakkara) [reply features/admin.branding.v1/components/screen-variation-dropdown.tsx:72]: addressed with https://github.com/wso2/identity-apps/pull/6027/commits/6f514ee32c24be59d56b0f82cd4c13abfa87c27b

File: .gitignore (hunk 5 above)
Comment (RushanNanayakkara) [reply .gitignore]: addressed with [cb35632](https://github.com/wso2/identity-apps/pull/6027/commits/cb35632c79a886962c53ebf5a56d90a028f6e909)

**Review by pavinduLakshan — APPROVED** [review]

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**